backend/sirius-components-collaborative-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/collaborative/formdescriptioneditors/FormDescriptionEditorEventProcessor.java

```diff
@@ -0,0 +1,170 @@
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.formdescriptioneditors;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.sirius.components.collaborative.api.ChangeDescription;
+import org.eclipse.sirius.components.collaborative.api.ChangeKind;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationRefreshPolicy;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationRefreshPolicyRegistry;
+import org.eclipse.sirius.components.collaborative.api.ISubscriptionManager;
+import org.eclipse.sirius.components.collaborative.formdescriptioneditors.api.IFormDescriptionEditorContext;
+import org.eclipse.sirius.components.collaborative.formdescriptioneditors.api.IFormDescriptionEditorCreationService;
+import org.eclipse.sirius.components.collaborative.formdescriptioneditors.api.IFormDescriptionEditorEventHandler;
+import org.eclipse.sirius.components.collaborative.formdescriptioneditors.api.IFormDescriptionEditorEventProcessor;
+import org.eclipse.sirius.components.collaborative.formdescriptioneditors.api.IFormDescriptionEditorInput;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IInput;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.core.api.IRepresentationDescriptionSearchService;
+import org.eclipse.sirius.components.core.api.IRepresentationInput;
+import org.eclipse.sirius.components.formdescriptioneditors.FormDescriptionEditor;
+import org.eclipse.sirius.components.formdescriptioneditors.description.FormDescriptionEditorDescription;
+import org.eclipse.sirius.components.representations.IRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * Reacts to the input that target the property sheet of a specific object and publishes updated versions of the
+ * {@link Form} to interested subscribers.
+ *
+ * @author arichard
+ */
+public class FormDescriptionEditorEventProcessor implements IFormDescriptionEditorEventProcessor {
+    private final Logger logger = LoggerFactory.getLogger(FormDescriptionEditorEventProcessor.class);
+
+    private final IEditingContext editingContext;
+
+    private final IFormDescriptionEditorContext formDescriptionEditorContext;
+
+    private final List<IFormDescriptionEditorEventHandler> formDescriptionEditorEventHandlers;
+
+    private final ISubscriptionManager subscriptionManager;
+
+    private final IFormDescriptionEditorCreationService formDescriptionEditorCreationService;
+
+    private final IRepresentationDescriptionSearchService representationDescriptionSearchService;
+
+    private final IRepresentationRefreshPolicyRegistry representationRefreshPolicyRegistry;
```

Comment:
A graphql subscribtion will be added in the next commit

Reply:
Sure but I don't see how the concept of `IRepresentationRefreshPolicyRegistry` would ever be of any use for this representation since it is not configurable 😄 

It has been create for representations which may end up displaying non semantic elements. For example, if a specifier creates a diagram to display remote data or things like that.